src/net_processing.cpp

```diff
@@ -4787,12 +4959,56 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                             pto->m_tx_relay->filterInventoryKnown.insert(txid);
                         }
                     }
+
+                    // Populating local reconciliation set.
+                    if (recon_state && txs_to_reconcile.size() != 0) {
+                        std::vector<uint256> txs_to_flood;
+                        int32_t recon_set_overflow = recon_state->GetLocalSetSize(false) + txs_to_reconcile.size() - MAX_PEER_TX_ANNOUNCEMENTS;
+                        if (recon_set_overflow > 0) {
+                            LogPrint(BCLog::NET, "Reconciliation set for the peer=%d is at capacity, not adding %i new transactions. \n", pto->GetId(), recon_set_overflow);
+                            // Since we reconcile frequently, it either means:
+                            // (1) a peer for some reason does not request reconciliations from us for a long while, or
+                            // (2) really a lot of valid fee-paying transactions were dumped on us at once.
+                            // We don't care about a laggy peer (1) because we probably can't help them even if we flood transactions.
+                            // However, exploiting (2) should not prevent us from relaying certain transactions.
+                            // Since computing a sketch over too many elements is too expensive, we will just flood some transactions here.
+                            txs_to_flood = std::vector<uint256>(txs_to_reconcile.end() - recon_set_overflow, txs_to_reconcile.end());
+                            txs_to_reconcile.resize(txs_to_reconcile.size() - recon_set_overflow);
+                        }
+                        // Do not use recon_state here because it's meant for read-only and
```

Comment:
I'm a little concerned about this thing here, but I'm really unsure how to make it better. Do you think it's fine? That's why we have const modifier, right? :)